Read credentials from command line

1ec5 opened the pull request:
If the user name and/or password is missing from the configuration file, read them from the command line. This change removes the need to specify the password in plain text in a file that’s relatively easy to find.

1ec5 committed:
Read credentials from command line

If the user name and/or password is missing from the configuration file, read them from the command line.

woodpeck commented:
I'm not sure this is a good idea. Many typical workflows involving these scripts consist of running various scripts repeatedly. If one was concerned about having the password in a file, would it perhaps be better to read it from an environment variable? That way the password could be used in a number of consecutive script executions.

1ec5 commented:
I personally use these scripts manually and prefer to keep my password out of my environment or history, but I can see the benefit in helping more automated workflows avoid putting the password in a file. What if the script were to read from the environment, then fall back to the .osmtoolsrc file, then finally fall back to reading user input?

woodpeck commented:
Sounds like a good plan. Is it possible to include the Term::ReadKey dependency in a way that it only kicks in when the password cannot be obtained elsewhere? That way users who don't intend to use the interactive facility don't have to install the module.

1ec5 committed:
Only use ReadKey if needed for prompts

1ec5 committed:
Look for credentials in environment variables

Prefer credentials from the environment, falling back to .osmtoolsrc and finally the prompts as a last resort. Updated the readme.

1ec5 commented:
OK, ab4c9e800516bb77cf5d94fd38ead563751f88c4 only uses `Term::ReadKey` if needed, and d9fe1bb39735152a036966f547e50ce873ae814d reads the credentials from environment variables, overriding whatever was in .osmtoolsrc.

woodpeck commented:
Great work, thank you, I merged this with a small fix.
